plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/polaris/PolarisHiveMetastore.java

```diff
@@ -0,0 +1,575 @@
+import io.trino.metastore.HivePrivilegeInfo;
+import io.trino.metastore.HiveType;
+import io.trino.metastore.Partition;
+import io.trino.metastore.PartitionStatistics;
+import io.trino.metastore.PartitionWithStatistics;
+import io.trino.metastore.PrincipalPrivileges;
+import io.trino.metastore.StatisticsUpdateMode;
+import io.trino.metastore.StorageFormat;
+import io.trino.metastore.Table;
+import io.trino.metastore.TableInfo;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.function.LanguageFunction;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.security.RoleGrant;
+import io.trino.spi.statistics.ColumnStatisticType;
+import io.trino.spi.type.Type;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.hive.HiveErrorCode.HIVE_METASTORE_ERROR;
+import static java.util.Objects.requireNonNull;
+
+public class PolarisHiveMetastore
+        implements HiveMetastore
+{
+    private final PolarisRestClient polarisClient;
+
+    @Inject
+    public PolarisHiveMetastore(PolarisRestClient polarisClient)
+    {
+        this.polarisClient = requireNonNull(polarisClient, "polarisClient is null");
+    }
+
+    @Override
+    public Optional<Database> getDatabase(String databaseName)
+    {
+        try {
+            PolarisNamespace namespace = polarisClient.getNamespace(databaseName);
+            return Optional.of(Database.builder()
+                    .setDatabaseName(namespace.getName())
+                    .setParameters(namespace.getProperties())
+                    .build());
+        }
+        catch (PolarisNotFoundException e) {
+            return Optional.empty();
+        }
+        catch (Exception e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, "Failed to get database: " + databaseName, e);
+        }
+    }
+
+    @Override
+    public List<String> getAllDatabases()
+    {
+        try {
+            return polarisClient.listNamespaces(Optional.empty())
+                    .stream()
+                    .map(PolarisNamespace::getName)
+                    .collect(toImmutableList());
+        }
+        catch (Exception e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, "Failed to list databases", e);
+        }
+    }
+
+    @Override
+    public Optional<Table> getTable(String databaseName, String tableName)
+    {
+        // Try Iceberg tables first
+        try {
+            if (polarisClient.tableExists(databaseName, tableName)) {
+                PolarisTableMetadata metadata = polarisClient.loadTable(databaseName, tableName);
+                return Optional.of(convertIcebergToHiveTable(databaseName, tableName, metadata));
+            }
+        }
+        catch (Exception e) {
```

Comment:
I think you need to replace `e` with `_` (or maybe ignored) to let the static checkers know you are ignoring this on purpose.

Also, `Exception` is too broad.  Use specific exception, maybe `PolarisNotFoundException`